src/data_analysis.py

```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+
+
+df = pd.read_csv('data/task3_dateset.csv', dtype=str)
+df['Date'] = pd.to_datetime(df['Date'], format='%m/%d/%y')
+df['Ads_Run'] = df['Ads_Run'].fillna(0)
+df['Ads_Run'] = df['Ads_Run'].astype(int)
+df = df.sort_values(by=['Site', 'Date'])
+
+
+def fix(name, df):
+    '''
+    fix the dataframe
+    '''
+
+    # renumber indexes starting at zero
+    df = df.reset_index(drop=True)
+
+    # make an "index" column
+    df = df.reset_index()
+
+    if not (df['Ads_Run'] != 0).any():
+        raise Exception(f'for {name}, we cannot infer a value')
+
+    sample = df[df['Ads_Run'] != 0].head(1)
+    offset = sample.index.values[0] - sample['Ads_Run'].values[0]
+    fix_idx = (df['index'] + offset) != df['Ads_Run']
```

Comment:
Note: could not just check for zero values because there may be values that are non-zero and wrong.